lib/net.c

```diff
@@ -1555,29 +1583,73 @@ start_network_stage2(void *arg)
 	return ret;
 }
 
+static int
+drop_privileges(void)
+{
+	int ret;
+	char user[128];
+
+	ret = getlogin_r(user, sizeof(user));
+	if (ret != 0) {
+		G_LOG(ERR, "%s: failed to get the current logged in user - %s!\n",
+			__func__, strerror(errno));
+		return ret;
+	}
+
+	ret = initgroups(user, config.pw_gid);
+	if (ret != 0) {
+		G_LOG(ERR, "%s: failed to call initgrous(%s, %u) - %s!\n",
+			__func__, user, config.pw_gid, strerror(errno));
+		return ret;
+	}
+
+	/* Drop privileges. */
+	if (setgid(config.pw_gid) != 0 || setuid(config.pw_uid) != 0) {
+		G_LOG(ERR, "%s: failed to drop privileges for user with user id %u and group id %u: %s!\n",
```

Comment:
Could you drop the exclamation mark from this log entry and all others in this patch?